File: Javascript/Linked List/M-2095. Delete the Middle Node of a Linked List.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function deleteMiddle(head: ListNode | null): ListNode | null {
  let node = head;
  let length = 0;
  while (node) {
    length++;
    node = node.next;
  }
  if (length === 1) return null;

  const middle = length >> 1;
  node = head;
  let count = 1;
  while (count < middle) {
    count++;
    node = node.next;
  }
  node.next = node.next?.next || null;
  return head;
}

File: Linked List/M-2095. Delete the Middle Node of a Linked List/M-2095. Delete the Middle Node of a Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        node = head
        node_length = 0
        while node:
            node_length += 1
            node = node.next

        if node_length == 1:
            return None

        middle = node_length // 2
        node = head
        count = 1
        while count < middle:
            node = node.next
            count += 1

        node.next = node.next.next if node.next is not None else None
        return head
